feat: implement sart instrument

rahul810050 opened the pull request:
Fixed Issue: #1211

Video Explanation: https://app.screencastify.com/watch/OMyACxMNXrvO63ta9u6G

rahul810050 committed:
feat: implement sart instrument

Signed-off-by: rahul810050 <rahul810050@gmail.com>

rahul810050 committed:
Update sartForm.vue

rahul810050 committed:
Update SartAnswer.js

rahul810050 committed:
Update TaskAnswer.js

rahul810050 committed:
Update sartData.js

rahul810050 commented:
@marcgc21 @KarinePistili @sahitya-chandra could you please review it ??

marcgc21 commented:
Instead of using the full string, use just SART:

<img width="1059" height="704" alt="Screenshot 2025-12-31 at 11 11 52" src="https://github.com/user-attachments/assets/7aad2db1-2fce-4bd6-a296-d4af0c8325d0" />

marcgc21 commented:
Delete SART summary, the user doesn't need to know what are the value because he/she is just answering:
<img width="1201" height="247" alt="Screenshot 2025-12-31 at 11 15 22" src="https://github.com/user-attachments/assets/1e1c9a36-3733-4918-9c60-d1def9b101b3" />

marcgc21 commented:
On SART analytics view it should be done some modifications to match the technique:

On top show:
- Average SA score
- Average Demand
- Average Supply
- Average Understanding

On SART dimension breakdown:
- Add a colored values for the bars
- Group bars on the three group dimensions
- Fit all the dimensions (right now, "Understanding" doesn't suit well).

On Individual SART Scores:
- Add a column called "Understanding"

On detail view analytics SART Score:
- Add the "Understanding Score"

rahul810050 commented:
Thanks for the clarification !! i will update the code according to your feedback @marcgc21

rahul810050 committed:
Merge branch 'develop' into feat/implement-sart-instrument

rahul810050 committed:
Update sartData.js

rahul810050 committed:
Update SartAnalytics.vue

rahul810050 committed:
Update TaskPreview.vue

rahul810050 commented:
@marcgc21 I have updated the code according to your feedback

After changing this is how things working...[vdo Link](https://app.screencastify.com/watch/431ssuIrSmlkc6X5EGVc)

rahul810050 committed:
Update sartForm.vue

rahul810050 commented:
@marcgc21 I have updated the code according to your feedback

After changing this is how things working...[vdo Link](https://app.screencastify.com/watch/431ssuIrSmlkc6X5EGVc)

marcgc21 commented:
I think that there are conflicts on the language files
<img width="781" height="423" alt="Screenshot 2026-01-01 at 12 21 52" src="https://github.com/user-attachments/assets/d673885a-693b-407f-b6e9-2373149ba33c" />

rahul810050 committed:
Update AnswerTypePreview.vue

rahul810050 committed:
Merge branch 'develop' into feat/implement-sart-instrument

rahul810050 committed:
Update en.json

rahul810050 commented:
@marcgc21 resolved the conflict. now it is fine

rahul810050 committed:
Update FormTask.vue

marcgc21 commented:
Still have problems. Please take your time to review correctly the PR before uploading again 🙏 

<img width="659" height="288" alt="Screenshot 2026-01-02 at 07 50 46" src="https://github.com/user-attachments/assets/5392663f-ad5c-47b1-9f95-06ec9de0e11f" />

rahul810050 commented:
@marcgc21 really sorry about that🙏 I will update the pr asap

rahul810050 committed:
Update UserTestAnswer.vue

rahul810050 commented:
@marcgc21 I have fixed the error. now it is working perfectly fine. once again really sorry about that mistake...

marcgc21 commented:
@rahul810050 we still have problems about SART methodology.
Idk where are you taken those dimensions, but "Supply" and "Understanding" are wrong:
<img width="922" height="282" alt="Screenshot 2026-01-02 at 14 18 11" src="https://github.com/user-attachments/assets/0451283f-64aa-4f33-a2b1-b33ad6728016" />

You can check this paper:
https://www.researchgate.net/figure/Situation-Awareness-Rating-Technique-SART-10D-Scale-30_fig9_365853339

Basically what it says is that Supply has 4 dimensions ant not 7:
- Arousal
- Concentration of Attention
- Division of Attention
- Spare Mental Capacity

And "Understanding" has:
- Information Quantity
- Information Quality
- Familiarity with Situation

So basically you are grouping Supply and Understanding. Try to fix this problem and I beliave everything will be done!

marcgc21 commented on src/ux/UserTest/utils/sartData.js:
```diff
@@ -0,0 +1,211 @@
+    category: 'supply'
+  },
+  { 
+    key: 'understanding', 
```
This should be the category "Understanding (U)"

marcgc21 commented on src/ux/UserTest/utils/sartData.js:
```diff
@@ -0,0 +1,211 @@
+    category: 'supply'
+  },
+  { 
+    key: 'informationQuantity', 
```
InformationQuantity, InformationQuality and Familiarity are from "Understanding" category

rahul810050 committed:
Update sartData.js

rahul810050 committed:
Update SartAnalytics.vue

rahul810050 committed:
Update sartForm.vue

rahul810050 committed:
Update SartAnswer.js

rahul810050 committed:
Update TaskStep.vue

rahul810050 commented:
@marcgc21 i have updated according to your feedback comments research paper. Just so you know i have changed the ui of sartForm a bit like put some colors according to dimensions just to make it a bit cool. if it seems wrong to you then i can make the ui as same as nasatlx form ui. let me know if you want exact that...

rahul810050 committed:
Update UserTestAnswer.vue

marcgc21 commented:
> ## [![Quality Gate Failed](https://camo.githubusercontent.com/92d3ccde9d97539f09151c26cc60b2de189e85bf5b7d33139c3ec0927dedf41a/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f5175616c6974794761746542616467652f71672d6661696c65642d323070782e706e67)](https://sonarcloud.io/dashboard?id=ruxailab_RUXAILAB&pullRequest=1233) **Quality Gate failed**
> Failed conditions ![](https://camo.githubusercontent.com/4fe73d190ee58e1cd813384cae342413435615102ab16a3fffc5c5ac2bb56044/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f6661696c65642d313670782e706e67) [4.1% Duplication on New Code](https://sonarcloud.io/component_measures?id=ruxailab_RUXAILAB&pullRequest=1233&metric=new_duplicated_lines_density&view=list) (required ≤ 3%)
> 
> [See analysis details on SonarQube Cloud](https://sonarcloud.io/dashboard?id=ruxailab_RUXAILAB&pullRequest=1233)

@rahul810050 can you take a look on that? Also we have fixed the lint action and it is reporting that the code is not fulfilling the requeriments. 

Can you also take a look on that? Regardind the feature I bet that it is already done, let's finish that and will finally merge it!

rahul810050 commented:
@marcgc21 i could be wrong but i raised this pr before the updated github actions that is why i think it still uses the previous github actions that is why it is failing....

for example  for this #1152  i raised a pr #1199 where i used the same logic  in useProfile.js(which fails in two actions same as this current pr) like duplicate staff that i also used in this pr #1266 in ProfileView.vue (but this is the latest develop branch code) and guess what it passes all the checks...i think you get the point marc!!! 

if you want i can again raise a new pr for this just to pass the github action let me know if you want that !!!

marcgc21 commented:
Hi @rahul810050  I reviewed the lint and it looks like only appears the files that you have modified, so I beliave that it is using the new version

rahul810050 commented:
whole files got lint error I fixed all the files !!! @marcgc21 if you dont want that i will revert these changes

marcgc21 commented:
@rahul810050 the problem now is that you applied the lint to the whole project... You need to do it just to the files that you have modified. So please revert the commit to the proper point of time and update it.

rahul810050 committed:
fix: auto-fix lint formatting for SART files

rahul810050 committed:
fix: remove side effects from computed properties in ModeratedTestView.vue

rahul810050 commented:
@marcgc21 i have reverted the changes.  please feel free to review it

marcgc21 committed:
fix(user-test): updated color logic and improved ux to visualize dimensions comparison by category

marcgc21 committed:
Merge branch 'develop' into feat/implement-sart-instrument
